packages/react-next/src/components/Pivot/useOverflow.ts

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import { getWindow } from '../../Utilities';
+import { observeResize } from './observeResize';
+
+/**
+ * Callback to notify the user that the items in the overflow have changed. This should ensure that the overflow menu
+ * is visible, and contains all of the overflowing items.
+ *
+ * @param overflowIndex The index of the first item in the overflow, or items.length if nothing is overflowing.
+ * @param items The list of all items in the container, and whether that particular item is in the overflow
+ */
+export type OverflowItemsChangedCallback = (
+  overflowIndex: number,
+  items: { ele: HTMLElement; isOverflowing: boolean }[],
+) => void;
+
+/**
+ * Track whether any items don't fit within their container, and move them to the overflow menu.
+ *
+ * The overflow menu button must be the last sibling of all of the items that can be put into the overflow:
+ *
+ * <Container>
+ *  <Item /> // Index 0
+ *  <Item /> // Index 1
+ *  ...
+ *  <OverflowMenuButton />
+ * </Container>
+ *
+ * Items will be moved into the overflow menu from back to front, excluding pinned items.
+ *
+ * @param overflowMenuButtonRef - The button that will show the overflow menu. All items that will go into the overflow
+ * menu should be direct siblings and come before this menu button.
+ *
+ * @param onOverflowItemsChanged - Callback to notify the user that the items in the overflow have changed.
+ *
+ * @param rtl - True if the element containing overflowMenuButtonRef is in right-to-left order
+ *
+ * @param pinnedIndexes - Optional: List of item indexes that should never go into the overflow menu.
+ */
+export const useOverflow = (
+  overflowMenuButtonRef: React.RefObject<HTMLElement | undefined> | null | undefined,
+  onOverflowItemsChanged: OverflowItemsChangedCallback,
+  rtl: boolean,
+  pinnedIndexes?: readonly number[],

```

Comment:
I think it's better to change this to 1 param, that is `options: OverflowOptions`. then overtime, we can expand/shrink the params more easily
```
type OverflowOptions = { overflowMenuButtonRef ;  ... }
```